stubs/psycopg2/psycopg2/_psycopg.pyi

```diff
@@ -303,10 +304,7 @@ class connection:
     def cancel(self, *args, **kwargs): ...
     def close(self, *args, **kwargs): ...
     def commit(self, *args, **kwargs): ...
-    @overload
-    def cursor(self) -> _cursor: ...
-    @overload
-    def cursor(self, name=..., cursor_factory: Any = ..., withhold=...) -> Any: ...
+    def cursor(self, name=..., cursor_factory: Callable[..., _T_cur] = ..., scrollable=..., withhold=...) -> _T_cur: ...
```

Comment:
We still need an overload for when `cursor_factory` is not provided, as otherwise `_T_cur` has an unknown type (despite being bound). Unfortunately that is a bit unwieldy (the original overloads did not cover all cases):

```suggestion
    @overload
    def cursor(self, name=..., *, scrollable=..., withhold=...) -> _cursor: ...
    @overload
    def cursor(self, name, cursor_factory: Callable[..., _T_cur], scrollable=..., withhold=...) -> _T_cur: ...
    @overload
    def cursor(self, *, cursor_factory: Callable[..., _T_cur], scrollable=..., withhold=...) -> _T_cur: ...
```

Reply:
Makes sense. I think we can do it with just two overload paths rather than three though.

Reply:
You are right, I was thinking too complex. There is now a theoretical overlap between the two overloads, but that shouldn't be an issue as overloads are processed in order.